spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/admin/deployment/zk/ZKDeploymentMessagePublisher.java

```diff
@@ -33,21 +83,164 @@
 	 * @param deploymentQueue the deployment queue
 	 */
 	public ZKDeploymentMessagePublisher(DeploymentQueue deploymentQueue) {
-		this.distributedQueue = deploymentQueue.getDistributedQueue();
+		this.deploymentQueue = deploymentQueue;
+	}
+
+	private CuratorFramework getClient() {
+		return this.deploymentQueue.getClient();
 	}
 
 	/**
-	 * Produces the deployment message into ZK distributed queue.
+	 * Return the amount of time in milliseconds to wait for the message
+	 * consumer to process the deployment message. This value is 3 times
+	 * the value of the configured module deployment timeout.
 	 *
-	 * @param deploymentMessage the deployment message
+	 * @see #deploymentTimeout
+	 * @return timeout for deployment message processing
 	 */
+	private long getTimeout() {
+		return deploymentTimeout * 3;
```

Comment:
While I am at it, I see a caveat when using deployment properties to determine the module count.  If the count is specified as `module.*.count` then no way we know ho many number of modules were there. One rough solution would be to get the number of pipes from the definition string and calculate the number of modules. This would work even if we have composed module in the definition string since we are concerned about the number of `ModuleDescriptors` that will be used for the deployment. What do you think?